docs/arch/adr-30.md

```diff
@@ -0,0 +1,133 @@
+
+More complex dropdown tasks will be built in the future as separate task types based on analysis of actual usage in PFE. These will include:
+
+- Cascading dropdowns for
+  - Locations
+  - Custom (TBD)
+- Date picker
+  - Possibly a text input with validation rather than dropdowns
+- Asynchronous loading long lists by text input search
+
+### Proposed task and annotation JSON structure examples
+
+**Task**
+``` json
+{
+  "T0":{
+    "help": "",
+    "type": "dropdown-simple",
+    "options":[
+      "Red",
+      "Blue",
+      "Green",
+      "Yellow"
+    ],
+    "required": false,
+    "allowCreate": false,
+    "instruction": "Choose your favourite colour"
+  }
+}
+```
+
+**Annotation**
+```json
+{
+  "annotations": [
+    {
+      "task":"T0",
+      "value": {
+        "value": 1,
+        "option": true
+      }
+    }
+  ]
+}
+```
+
+## Consequences
+ 
+
+Theses decisions allow for these improvements:
+
+- Project owners will have the actual index of the values selected from the dropdown task in either the raw classification or aggregation exports. This is just like the single choice task.
+- The aggregation code won’t have to dynamically choose what kind of aggregation to perform since the task type will be solely dropdown and not a combination of dropdown and text entry.
+- Min and max options allow us to avoid performance and load issues.
+
+However these decisions have impact on building the task in the lab and may mean dev time in the future:
+
+- If an option isn’t on the dropdown list, then the research team will need to include a separate text task in the workflow step. Note that the team has the possibility to add the word “Other” to the options list if they like, but a blank entry can serve this purpose as well. We can reevaluate this decision with more data on actual usage of the "option" feature in PFE and with feedback from project builders. To this end, the task and annotation structure will keep the `option` in the annotation and `allowCreate` in the task properties for now in case the feedback is overwhelmingly that project builders really need this configurability. 
+- Dates will need to be set up manually, e.g. free-text for Day, researcher-generated list of months for Month. Note that for Engaging Crowds project #1, each subject set will cover the same year, so that info should already be in the subject metadata. In the future, we will develop some kind of date input task, but it won’t be available for the initial Engaging Crowds effort.
+- Generally project owners using the PFE dropdown task will need to migrate to one of the new options, and thus there will not be backwards compatibility with the PFE version.
+- The updated task type will need an updated editor in the lab.
```

Comment:
I think the new task type will also need translatable strings defined in Panoptes, so that the task question and options are available in the workflow translations editor.

Reply:
Here's an example of the translation keys for a single dropdown with the PFE task. I think the new task will simplify the keys.
https://pandora.zooniverse.org/#/project/908/workflow/3339?language=es

![Spanish translations for a single dropdown menu, showing the key names for the task help, instruction and option labels.](https://user-images.githubusercontent.com/59547/104918990-7994fc80-598d-11eb-98f8-1da688329c32.png)